metadata-ingestion/src/datahub/ingestion/source/sql/postgres.py

```diff
@@ -144,13 +150,11 @@ def create(cls, config_dict, ctx):
         return cls(config, ctx)
 
     def get_inspectors(self) -> Iterable[Inspector]:
-        # This method can be overridden in the case that you want to dynamically
-        # run on multiple databases.
-        url = self.config.get_sql_alchemy_url()
+        url = self.config.get_sql_alchemy_url(database=self.config.init_database)
         logger.debug(f"sql_alchemy_url={url}")
         engine = create_engine(url, **self.config.options)
         with engine.connect() as conn:
-            if self.config.database and self.config.database != "":
+            if self.config.database or self.config.sqlalchemy_uri:
                 inspector = inspect(conn)
```

Comment:
This doesn't seem to be right.
When you reach here, `conn` is using `self.config.init_database`, but you want to connect using `self.config.database` or `self.config.sqlalchemy_uri`.

Reply:
So we changed the logic here to be more backwards compatible with the previous logic, basically:
- If `database` or `sqlalchemy_uri` are provided, we _only_ connect to that database / that uri. Thus, if either are provided, `init_database` and `database_pattern` are ignored, and we only ingest from a single database.
- If neither are provided, then we're ingesting multiple databases. We initially connect to `init_database` to query for the list of databases, then ingest from each that matches `database_pattern`.

This logic is largely handled by the `config.get_sql_alchemy_url` method.
I'm hoping the added notes on the docs for `database_pattern` and `init_database` will make this behavior clear.

Reply:
Ah, got it.